app/scripts/lib/app-start.js

```diff
@@ -302,56 +302,32 @@ define(function (require, exports, module) {
+        let iframeChannel = this._iframeChannel;
+        let metrics = this._metrics;
+        let oAuthClient = this._oAuthClient;
+        let relier = this._relier;
+        let session = Session;
+        let window = this._window;
+
         if (this._isFxFirstrunV2()) {
-          this._authenticationBroker = new FxFirstrunV2AuthenticationBroker({
-            iframeChannel: this._iframeChannel,
-            relier: this._relier,
-            window: this._window
-          });
+          this._authenticationBroker = new FxFirstrunV2AuthenticationBroker({iframeChannel,relier,window});
         } else if (this._isFxFirstrunV1()) {
-          this._authenticationBroker = new FxFirstrunV1AuthenticationBroker({
-            iframeChannel: this._iframeChannel,
-            relier: this._relier,
-            window: this._window
-          });
+          this._authenticationBroker = new FxFirstrunV1AuthenticationBroker({iframeChannel,relier,window});
         } else if (this._isFxFennecV1()) {
-          this._authenticationBroker = new FxFennecV1AuthenticationBroker({
-            relier: this._relier,
-            window: this._window
-          });
+          this._authenticationBroker = new FxFennecV1AuthenticationBroker({relier,window});
         } else if (this._isFxDesktopV3()) {
-          this._authenticationBroker = new FxDesktopV3AuthenticationBroker({
-            relier: this._relier,
-            window: this._window
-          });
+          this._authenticationBroker = new FxDesktopV3AuthenticationBroker({relier,window});
         } else if (this._isFxDesktopV2()) {
-          this._authenticationBroker = new FxDesktopV2AuthenticationBroker({
-            relier: this._relier,
-            window: this._window
-          });
+          this._authenticationBroker = new FxDesktopV2AuthenticationBroker({relier,window});
         } else if (this._isFxDesktopV1()) {
-          this._authenticationBroker = new FxDesktopV1AuthenticationBroker({
-            relier: this._relier,
-            window: this._window
-          });
+          this._authenticationBroker = new FxDesktopV1AuthenticationBroker({relier,window});
         } else if (this._isFxiOSV1()) {
-          this._authenticationBroker = new FxiOSV1AuthenticationBroker({
-            relier: this._relier,
-            window: this._window
-          });
+          this._authenticationBroker = new FxiOSV1AuthenticationBroker({relier,window});
         } else if (this._isOAuth()) {
-          this._authenticationBroker = new RedirectAuthenticationBroker({
-            assertionLibrary: this._assertionLibrary,
-            metrics: this._metrics,
-            oAuthClient: this._oAuthClient,
-            relier: this._relier,
-            session: Session,
-            window: this._window
-          });
+          this._authenticationBroker = new RedirectAuthenticationBroker({assertionLibrary,metrics,oAuthClient,relier,session,window});
```

Comment:
this line seems long, can we do something this for all:

```js
({
  assertionLibrary,
  metrics,
  oAuthClient,
  relier,
  session,
  window
})
```

^ this way it is easier to read